File: easy_1/series/ruby/rework/series.rb
# series.rb

class Series
  def initialize(input_series)
    self.series = input_series
  end

  def slices(n)
    raise ArgumentError if n > series.length

    substrs = substrings_of_length(n)
    substrings_to_ints(substrs)
  end

  private
  attr_accessor :series

  def substrings_of_length(n)
    pointer = 0
    substrs = []

    while (pointer + n) <= series.length
      substrs << series.slice(pointer, n)
      pointer += 1
    end

    substrs
  end

  def substrings_to_ints(substrs)
    substrs.map do |substr|
      ints = []
      substr.each_char { |char| ints << char.to_i }
      ints
    end
  end
end

File: easy_1/series/js/series.js
// series.js

export default class Series {
  constructor(series) {
    this.series = series;
  };

  slices(n) {
    if (n > this.series.length) {
      throw "Invalid slice length.";
    }

    const substrs = this.substringsOfLength(n);
    return this.substrsToNums(substrs);
  };

  substringsOfLength = (n) => {
    let pointer = 0;
    const substrs = [];

    while((pointer + n) <= this.series.length) {
      substrs.push(this.series.substr(pointer, n));
      pointer += 1;
    }

    return substrs;
  };

  substrsToNums = (substrs) => {
    return substrs.map( (substr) => {
      const nums = [];

      for(let idx = 0; idx < substr.length; idx += 1) {
        const char = substr[idx];
        nums.push(Number(char));
      };

      return nums;
    });
  };
};